plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreAuthenticationModule.java

```diff
@@ -35,29 +36,28 @@
     @Override
     protected void setup(Binder binder)
     {
-        install(getAuthenticationModule());
-    }
+        install(new OptionalDefaultHiveMetastoreAuthenticationModule());
 
-    private Module getAuthenticationModule()
-    {
         ThriftMetastoreAuthenticationType type = buildConfigObject(ThriftMetastoreAuthenticationConfig.class).getAuthenticationType();
         switch (type) {
             case NONE:
-                return new NoHiveMetastoreAuthenticationModule();
+                // already bound optionally
+                break;
             case KERBEROS:
-                return new KerberosHiveMetastoreAuthenticationModule();
+                install(new KerberosHiveMetastoreAuthenticationModule());
         }
-        throw new AssertionError("Unknown authentication type: " + type);
```

Comment:
Is the goal to allow additional types in the future?  If so, I'd rewrite this as:
```java
if (buildConfigObject(ThriftMetastoreAuthenticationConfig.class).getAuthenticationType() == KERBEROS) {
    install(new KerberosHiveMetastoreAuthenticationModule());
}